misc/bazel/transitions.bzl

```diff
@@ -0,0 +1,141 @@
+load("@bazel_skylib//lib:paths.bzl", "paths")
+
+def _make_platform_transition(impl):
+    return transition(
+        implementation = impl,
+        inputs = ["//command_line_option:platforms"],
+        outputs = ["//command_line_option:platforms"],
+    )
+
+def _platform_transition_impl(_settings, attr):
+    return {
+        "//command_line_option:platforms": [attr.platform],
+    }
+
+# Transition to attr.platform
+platform_transition = _make_platform_transition(_platform_transition_impl)
+
+def _get_platform_for_arch(platform, arch):
+    # temporary, for backward compatibility with internal repo while we move stuff around
+    if ("cc_configure_toolchain//:default_platform" in platform or
+        "//toolchains/platforms" in platform):
+        if "host_toolchain" in platform:
+            return "@semmle_code//toolchain/platforms:host_toolchain_%s" % arch
+        return "@semmle_code//toolchain/platforms:bundled_toolchain_%s" % arch
+    if "bundled_toolchain" in platform:
+        return "@codeql//misc/bazel/platforms:bundled_toolchain_%s" % arch
+
+    # this default also covers bazel's own default
+    return "@codeql//misc/bazel/platforms:host_toolchain_%s" % arch
+
+def _x86_32_transition_impl(settings, _attr):
```

Comment:
This is now a lot of stuff we're only using internally (as we don't have C++ code that ends up in a pack in the external repo). That introduces a coupling between internal+external repo we didn't have before. Would it make sense to split this up into the universal binary transition, and the rest?

I believe we already transitively depend on the `apple_support` (bazel itself does, iirc). Avoiding that dependency was the only reason not to use https://github.com/bazelbuild/apple_support/blob/master/doc/rules.md#universal_binary.
Maybe now's the time to drop our lipo rule and use that instead? That saves us from having to move some of this code around, and an external dep is always easiest from a repo coordination perspective.

Reply:
agreed

Reply:
> I believe we already transitively depend on the apple_support (bazel itself does, iirc). Avoiding that dependency was the only reason not to use https://github.com/bazelbuild/apple_support/blob/master/doc/rules.md#universal_binary.
Maybe now's the time to drop our lipo rule and use that instead? That saves us from having to move some of this code around, and an external dep is always easiest from a repo coordination perspective.

I don't think we rely implicitly on `apple_support`, at least it does not appear in `bazel mod graph`. The big downside we found in `apple_support` and its `universal_binary`, is that it relied on a full Xcode installation (I think it calls `xcrun lipo`), and we really don't want such a heavy dependency for our developers when we can use command line tools. I don't think that has changed in the meantime.